packages/terra-navigation-prompt/docs/README.md

```diff
@@ -0,0 +1,138 @@
+
+#### `onPromptChange`
+
+The `onPromptChange` function prop is used to communicate NavigationPrompt registrations to the implementer of the NavigationPromptCheckpoint. The function is called with an array of objects containing each registered NavigationPrompt's `description` and `metaData` properties.
+
+> Note: The `onPromptChange` prop will be called after each NavigationPrompt registration and removal. It may be called more than once with the same array of NavigationPrompt data. Calls to `setState` within onPromptChange should be performed only when necessary to minimize component updates.
+
+#### `resolvePrompts`
+
+The `resolvePrompts` function be accessed from the ref to a NavigationPromptCheckpoint. Calling `resolvePrompts` results in a Promise being returned and a NotificationDialog being presented to the user with options to either confirm or cancel their action. If the user confirms the action, the dialog will close, and the returned Promise will be resolved. If the user cancels the action, the dialog will close, and the returned Promise will be rejected. If no NavigationPrompts are detected, no dialog is presented, and a resolved Promise is returned.
+
+`resolvePrompts` accepts a single Object argument that should contain the text strings used to render the NotificationDialog. If these strings need to present information based on the currently registered NavigationPrompts, the `onPromptChange` prop can be used to receive that data and build those strings as needed.
+
+The supported keys in the resolvePropts argument include:
+
+|Key|Type|Description|
+|---|---|---|
+|`title`|String|The title of the NotificationDialog|
+|`message`|String|The message of the NotificationDialog|
+|`rejectButtonText`|String|The text to render within the NotificationDialog's primary action button|
+|`acceptButtonText`|String|The text to render within the NotificationDialog's secondary action button|
+
+```jsx
+constructor() {
+  this.checkpointRef = React.createRef();
+  this.state = {
+    contentKey: false,
+  }
+}
+
+render() {
+  const { contentKey } = this.state;
+  return (
+    <div>
+      <button
+        onClick={() => {
+          this.checkpointRef.current.resolvePrompts({
+            title: 'Pending Changes',
+            message: 'Content data will be lost if this action is taken.',
+            rejectButtonText: `Return`,
+            acceptButtonText: 'Continue without Saving',
+          }).then(() => {
+            // User decided to continue
+            this.setState(state => ({ contentKey: !state.contentKey }));
+          }).catch(() => {
+            // User decided to cancel, so the state is not updated.
+          });
+        }}
+      >
+        Reset Content
+      </button>
+      <NavigationPromptCheckpoint
+        ref={this.checkpointRef}
+      >
+        <Content key={contentKey} />
+      </NavigationPromptCheckpoint>
+    </div>
+  );
+}
+```
+
+### `PromptRegistrationContext`
```

Comment:
This is the first mention of PromptRegistrationContext. As a consumer, should I be using this or is this just to explain how this works behind the scenes? I'm not seeing reference in the guides

Reply:
I waffled on including this at all, but I figured I should document this for anyone wanting to interface with the context themselves. I can further document my expectations around this.

Reply:
I waffled some more. I'm just going to remove it. I don't want to open the door to that level of customization out of the gate.